### PR TITLE
Refactor howto import data documentation.

### LINKED ISSUE #197: Document how to bulk import/export data
FAQ: How do I bulk import resources?

Before bulk importing resources, let's first review the resource table. The tables in ColdFront are managed by the Django ORM. The ORM model definitions, which define the resource table and related tables, are defined [here](https://github.com/ubccr/coldfront/blob/master/coldfront/core/resource/models.py.  

A resource has the following required columns:
1. `resource_type` -- this column relates the `Resource` table to the `ResourceType` table. The `ResourceType`  table defines the types of resources that can be added to ColdFront. By default, ColdFront adds the following resource types: `Cloud`, `Cluster`,  `Cluster Partition`,  `Server`,  `Software License`, and `Storage`.  Other resource types can easily be added through the ColdFront admin site.
2. `name` -- name of the resource
3. `description` -- description of the resource

A resource can have the following optional columns:
1. `parent_resource` -- [default: Null] this column relates a resource to another resource that is its parent. This relationship can model a scenario, for example, where a cluster has many partitions where the parent resource itself is not allocatable, but its partitions are. 
2. `is_available` -- [default: True] a boolean to change the availability of a resource. Useful to disable users from requesting allocation to the resource.
3.`is_public` -- [default: True] a boolean indicating if everyone can request allocation to the resource or not. 
4. `is_allocatable` -- [default: True] -- whether the resource is allocatable or not. This column can be used in conjunction with the parent_resource column to allow users to request access to children resources and not to parent resources. 
5. `requires_payment` -- [default: False] -- used to indicate if the resource requires payment.
6. `allowed_groups` -- a many-to-many column that allows users to request access to the resource if they belong in a group that in the allowed_groups column. This column is useful when the `is_public` column is set to False. 
7. `allowed_users` -- a many-to-many column that allows users to request access to the resource if they are in this list of allowed users. This column is useful when the `is_public` column is set to False. 
8. `linked_resources` -- a -many-to-many column that automatically gives allocation to resources in the linked_resources column in addition to the resource itself. 

To bulk import resources, first create a spreadsheet with the resource information and export it as TAB delimited file. The following code block shows a list of resources with the columns that the importing script requires. NOTE: the tabs get lost when copying and pasting. Also some editors convert tabs to spaces. So be careful!

```
#name	description	resource_type	parent_resource	is_available	is_public	is_allocatable
Faculty Cluster	John's private cluster	Cluster		TRUE	FALSE	TRUE
General Cluster	Public Cluster	Cluster		TRUE	TRUE	TRUE
Test Cluster Main	Main Cluster	Cluster		TRUE	FALSE	FALSE
Test Cluster Partition1	Partition 1 of test cluster	Cluster Partition	Test Cluster Main	TRUE	TRUE	TRUE
Test Cluster Partition2	Partition 2 of test cluster	Cluster Partition	Test Cluster Main	TRUE	TRUE	TRUE
```

Drop the following piece of code in `coldfront/coldfront/core/utils/management/commands/import_resources_from_tsv.py`

```
import errno
import os

import argparse
parser = argparse.ArgumentParser()


from django.core.management.base import BaseCommand

from coldfront.core.resource.models import ResourceType
from coldfront.core.resource.models import Resource


class Command(BaseCommand):

    def add_arguments(self, parser):
        parser.add_argument(
            "-f", "--file", help="""Path to a tab seperated file with the following columns:
            name, description, resource_type, parent_resource, is_available, is_public, is_allocatable.""",
            type=str, required=True)

    def handle(self, *args, **options):

        filename = options['file']
        if not os.path.isfile(filename):
            raise FileNotFoundError(
                errno.ENOENT, os.strerror(errno.ENOENT), filename)

        with open(filename, 'r') as fh:
            for line in fh:

                if line.startswith('#'):
                    continue
                name, description, resource_type, parent_resource, is_available, is_public, is_allocatable = line.strip().split('\t')

                if is_available.strip().lower() == 'true':
                    is_available = True
                else:
                    is_available = False

                if is_public.strip().lower() == 'true':
                    is_public = True
                else:
                    is_public = False

                if is_allocatable.strip().lower() == 'true':
                    is_allocatable = True
                else:
                    is_allocatable = False

                resource_obj, created = Resource.objects.get_or_create(
                    name=name.strip(),
                    description=description.strip(),
                    resource_type=ResourceType.objects.get(
                        name=resource_type.strip()),
                    is_available=is_available,
                    is_public=is_public,
                    is_allocatable=is_allocatable
                )

                if parent_resource.strip():
                    parent_resource = Resource.objects.get(
                        name=parent_resource.strip())
                    resource_obj.parent_resource = parent_resource
                    resource_obj.save()

                print(resource_obj)
```
 

Import the resources: 

```
python3 manage.py import_resources_from_tsv --file <location_to_file>
```

### PR DESCRIPTION
**Pull request by aebruno**
This commit refactors bulk data import documentation. Fixes #197  replaces #506

### CODE REVIEW

File: docs/pages/manual/howto/importing.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+]
+```
+
+To load the json file we use the Django (ColdFront) `loaddata` command:
+
+```coldfront loaddata --format=json allocations.json```
+
+## Importing other data
+
+The previous sections provide examples of import specific data into ColdFront.
```
Comment (dsajdak) [inline docs/pages/manual/howto/importing.md:201]: recommend changing 'import' to 'importing'

**Review by dsajdak — CHANGES_REQUESTED** [review]
Just a few minor changes, in addition to adding the PI as a user on the project and allocation

File: docs/pages/manual/howto/importing.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
+# Importing Data
+
+This howto guide demonstrates importing data into ColdFront using json files.
+The examples are very basic and can be easily modified to import data from any
+legacy system. 
+
+The following sections describe how you can builk import Users, Projects,
```
[the remark this answers is not in the document]
Comment (aebruno) [reply docs/pages/manual/howto/importing.md]: fixed

**Review by dsajdak — APPROVED** [review]
Just one minor typo but other than that this all works well.

File: docs/pages/manual/howto/importing.md [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+        allocation=a,
+        user=piuser,
+        status=status
+    )
+```
+
+Then you can run the above python code using the Django (ColdFront) shell:
+
+```
+$ coldfront shell < import-alloc-users.py`
```
Comment (dsajdak) [inline docs/pages/manual/howto/importing.md:329]:
```suggestion
$ coldfront shell < import-alloc-users.py

```

**Pull request merged**